### PR TITLE
add user index_spec

### PR DESCRIPTION
**Pull request by leloufadel**
In this branch:

We fixed the N+1 problem is solved when fetching all posts and their comments for a user
**Screenshot before the fix** 
TEST-bfore-includes
 **screenshot after fixing**                            
TEST-after-includes
 User index page:
We can see the username of all other users.
We can see the profile picture for each user.
We can see the number of posts each user has written.
When we click on a user, I am redirected to that user's show page.
User show page:
We can see the user's profile picture.
We can see the user's username.
We can see the number of posts the user has written.
We can see the user's bio.
We can see the user's first 3 posts.
We can see a button that lets me view all of a user's posts.
When click a user's post, it redirects me to that post's show page.
When click to see all posts, it redirects me to the user's post's index page.
User post index page:
We can see the user's profile picture.
We can see the user's username.
We can see the number of posts the user has written.
We can see a post's title.
We can see some of the post's body.
We can see the first comments on a post.
We can see how many comments a post has.
We can see how many likes a post has.
We can see a section for pagination if there are more posts than fit on the view.
When click on a post, it redirects me to that post's show page.
Post show page:
We can see the post's title.
We can see who wrote the post.
We can see how many comments it has.
We can see how many likes it has.
We can see the post body.
We can see the username of each commenter.
We can see the comment each commentor left.

### CODE REVIEW

File: spec/features/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  end
+  scenario 'I can see the profile picture for each user' do
+    visit user_path(@user1)
+    expect(page).to have_css('img[src*="user1_profile_image.jpg"]')
+  end
+  scenario 'I can see the number of posts of the user' do
+    visit user_path(@user1)
+    expect(page).to have_content(@user1.postsCounter)
+  end
+end
```
Comment (ShadyShawkat) [inline spec/features/user_index_spec.rb]: - [ ] As per the requirements, kindly add [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=When%20I%20click%20on%20a%20user%2C%20I%20am%20redirected%20to%20that%20user%27s%20show%20page.) test case. (Kindly note that some of the tests should simulate user interaction using the `capybara` gem, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=Use%20Capybara%20to%20write%20integration%20tests%20for%20each%20view%20in%20your%20project.) requirement)

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    expect(page).to have_text(@post3.title)
+    expect(page).not_to have_text(@post4.title)
+    expect(page).not_to have_text(@post5.title)
+    expect(page).not_to have_text(@post6.title)
+  end
+
+  scenario 'When I click to see all posts, it redirects me to the user\'s post\'s index page' do
+    click_link 'See All Posts'
+    expect(current_path).to eq(user_posts_path(@user1))
+  end
```
Comment (ShadyShawkat) [inline spec/features/user_show_spec.rb]: - [ ] As per the requirements, kindly add [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=When%20I%20click%20a%20user%27s%20post%2C%20it%20redirects%20me%20to%20that%20post%27s%20show%20page.) test case. (Kindly note that some of the tests should simulate user interaction using the `capybara` gem, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=Use%20Capybara%20to%20write%20integration%20tests%20for%20each%20view%20in%20your%20project.) requirement)

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @leloufadel, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All tests are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TuyishimireEric — APPROVED** [review]
# Status: APPROVED 🟢  
### Hi  @leloufadel  👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉


## Highlight 👍 
✅ Made suggested changes 👍 

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**